src/StorageWriter/DAOIntegrityCheckFailures.ts

```diff
@@ -0,0 +1,23 @@
+import { SignedVerifiableClaim } from '@po.et/poet-js'
+import { inject, injectable } from 'inversify'
+import { Collection, Db} from 'mongodb'
+
+@injectable()
+export class DAOIntegrityCheckFailures {
+  private readonly collection: Collection
+
+  constructor(
+    @inject('DB') db: Db,
+  ) {
+    this.collection = db.collection('storageWriterIntegrityCheckFailures')
```

Comment:
Let's inject the collection directly and move `db.collection('storageWriterIntegrityCheckFailures')` to the DI container creation